lib/src/test/kotlin/starknet/data/TypedDataTest.kt

```diff
@@ -703,6 +711,34 @@ internal class TypedDataTest {
                 message = "{\"$includedType\": 1}",
             )
         }
+
+        @Test
+        fun `domain v1 object with integer revision`() {
+            val resolvedRevision = TypedData(
+                types = mapOf(
+                    domainTypeV1,
+                ),
+                primaryType = "StarknetDomain",
+                domain = domainObjectV1,
+                message = "{\"StarknetDomain\": 1}",
+            ).domain.resolvedRevision
+
+            assertEquals(Revision.V1, resolvedRevision)
+        }
+
+        @Test
+        fun `domain v1 object with string revision`() {
+            val resolvedRevision = TypedData(
+                types = mapOf(
+                    domainTypeV1,
+                ),
+                primaryType = "StarknetDomain",
+                domain = domainObjectV1WithStringRevision,
+                message = "{\"StarknetDomain\": 1}",
+            ).domain.resolvedRevision
+
+            assertEquals(Revision.V1, resolvedRevision)
+        }
```

Comment:
Nit: 
```suggestion
        @Test
        fun `domain v1 object with string revision`() {
            val td = TypedData(
                types = mapOf(
                    domainTypeV1,
                ),
                primaryType = "StarknetDomain",
                domain = domainObjectV1WithStringRevision,
                message = "{\"StarknetDomain\": 1}",
            )

            assertEquals(Revision.V1, resolvedRevision.domain.resolvedRevision)
        }
```

Just makes catching td creation errors more easy (hypothetically, very unlikely situation 😅)